core/cache/model.go

```diff
@@ -237,63 +224,32 @@ func (c *modelConfigHashCache) generateHash(keys []string) string {
 }
 
 type modelConfigWatcher struct {
-	keys    []string
-	hash    string
-	tomb    tomb.Tomb
-	changes chan struct{}
-	// We can't send down a closed channel, so protect the sending
-	// with a mutex and bool. Since you can't really even ask a channel
-	// if it is closed.
-	closed bool
-	mu     sync.Mutex
+	*notifyWatcherBase
+
+	keys []string
+	hash string
+}
+
+func newModelConfigWatcher(keys []string, keyHash string) *modelConfigWatcher {
+	sort.Strings(keys)
+
+	return &modelConfigWatcher{
+		notifyWatcherBase: newNotifyWatcherBase(),
+
+		keys: keys,
+		hash: keyHash,
+	}
 }
 
 func (w *modelConfigWatcher) configChanged(topic string, value interface{}) {
 	hashCache, ok := value.(*modelConfigHashCache)
```

Comment:
I am going to make this hash-cache and config watching more generic in a following patch, then apply it to `Application`.